src/EpiEasyEvents/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Forte.EpiEasyEvents;
+using Microsoft.AspNetCore.Builder;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class ServiceCollectionExtensions
+    {
+        public static IServiceCollection AddEpiEasyEvents(this IServiceCollection services, IEnumerable<Assembly> assemblies)
+        {
+            services.AddSingleton<EventsRegistry>();
+            RegisterEventHandlers(services, assemblies);
+
+            return services;
+        }
+
+        public static IServiceCollection AddEpiEasyEvents(this IServiceCollection services, Assembly assembly)
+        {
+            return services.AddEpiEasyEvents(new[] {assembly});
+        }
+
+        public static void UseEpiEasyEvents(this IApplicationBuilder app)
+        {
+            var easyEventsRegistry = app.ApplicationServices.GetRequiredService<EventsRegistry>();
+            easyEventsRegistry.RegisterEvents();
+        }
+
+        private static void RegisterEventHandlers(IServiceCollection services, IEnumerable<Assembly> assemblies)
+        {
+            var interfaceWithTypeTuples = assemblies.SelectMany(assembly => assembly.GetTypes())
+                .SelectMany(
+                    type => type.GetInterfaces()
+                        .Where(interfaceType => interfaceType.IsGenericType && interfaceType.GetGenericTypeDefinition() == typeof(IContentEventHandler<,>))
+                        .Select(it => (it, type)));
+
+            foreach (var (interfaceToRegister, typeToRegister) in interfaceWithTypeTuples)
+            {
+                services.AddTransient(interfaceToRegister, typeToRegister);
```

Comment:
Only to confirm as we register here more that one implementationType for serviceType
What is the desired behaviour if we have more than one implementation for of IContentEventHandler<ConreteType>? Are we able to handle/call one by one?

Reply:
yes, there are several possible scenarios: you might want to have multiple handlers of `IContentPublishedHandler<PageData>` (to rebuild search index, to set published date etc). In this scenario you have more than one class registered as an implementation of `IContentPublishedHandler<PageData> ` and all of these classes will be fetched and called in `EventsRegistry.HandleEvent` method